Newer versions of pylink are not supported in latest zephyr 2.7 release

**Describe the bug**
It is not possible to mix a zephyr 3.1.99 and zephyr 2.7.3 development environment on a single PC due to the code in zephyr 2.7 not supporting the renamed files in newer versions of pylink, when trying to run `ninja debug`:
```
-- west debug: rebuilding
ninja: no work to do.
-- west debug: using runner jlink
Traceback (most recent call last):
  File "/usr/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/home/jamie/.local/lib/python3.10/site-packages/west/__main__.py", line 3, in <module>
    main()
  File "/home/jamie/.local/lib/python3.10/site-packages/west/app/main.py", line 786, in main
    app.run(argv or sys.argv[1:])
  File "/home/jamie/.local/lib/python3.10/site-packages/west/app/main.py", line 109, in run
    self.run_command(argv)
  File "/home/jamie/.local/lib/python3.10/site-packages/west/app/main.py", line 343, in run_command
    self.run_extension(args.command, argv)
  File "/home/jamie/.local/lib/python3.10/site-packages/west/app/main.py", line 415, in run_extension
    command.run(args, unknown, self.topdir, manifest=self.manifest,
  File "/home/jamie/.local/lib/python3.10/site-packages/west/commands.py", line 135, in run
    self.do_run(args, unknown)
  File "/tmp/bb/zephyr/scripts/west_commands/debug.py", line 34, in do_run
    do_run_common(self, my_args, runner_args)
  File "/tmp/bb/zephyr/scripts/west_commands/run_common.py", line 228, in do_run_common
    runner.run(command_name)
  File "/tmp/bb/zephyr/scripts/west_commands/runners/core.py", line 509, in run
    self.do_run(command, **kwargs)
  File "/tmp/bb/zephyr/scripts/west_commands/runners/jlink.py", line 196, in do_run
    self.logger.info(f'JLink version: {self.jlink_version_str}')
  File "/tmp/bb/zephyr/scripts/west_commands/runners/jlink.py", line 171, in jlink_version_str
    major, minor, rev = self.jlink_version
  File "/tmp/bb/zephyr/scripts/west_commands/runners/jlink.py", line 155, in jlink_version
    lib = Library(dllpath=os.fspath(Path(self.commander).parent /
  File "/home/jamie/.local/lib/python3.10/site-packages/pylink/library.py", line 278, in __init__
    self.load(dllpath)
  File "/home/jamie/.local/lib/python3.10/site-packages/pylink/library.py", line 384, in load
    with open(self._path, 'rb') as inputfile:
FileNotFoundError: [Errno 2] No such file or directory: '/opt/SEGGER/JLink/jlinkarm.so'
FAILED: zephyr/cmake/flash/CMakeFiles/debug /tmp/bb/zephyr/samples/subsys/mgmt/mcumgr/smp_svr/_AA/zephyr/cmake/flash/CMakeFiles/debug 
cd /tmp/bb/zephyr/samples/subsys/mgmt/mcumgr/smp_svr/_AA && /usr/bin/cmake -DTARGET=debug -DDEPENDENCIES="" -P /tmp/bb/zephyr/cmake/flash/check_runner_dependencies.cmake && /usr/bin/cmake -E env /usr/bin/python3.10 -m west debug
ninja: build stopped: subcommand failed.
```

**To Reproduce**
Get zephyr main, install requirements.txt, get zephyr 2.7.3, build and debug a project using `ninja debug`

**Expected behavior**
For the command to work

**Impact**
Probably a bug impact for people trying to use both environments

**Environment (please complete the following information):**
 - OS: Linux
 - Toolchain unrelated
 - Commit SHA or Version used: 003de78ce0dd213a1c7b3d159b967fb19a12aa45

--- a/scripts/west_commands/runners/jlink.py
+++ b/scripts/west_commands/runners/jlink.py
@@ -7,6 +7,7 @@
 import argparse
 import logging
 import os
+from packaging.version import Version
 from pathlib import Path
 import shlex
 import subprocess
@@ -16,6 +17,7 @@ import tempfile
 from runners.core import ZephyrBinaryRunner, RunnerCaps
 
 try:
+    import pylink
     from pylink.library import Library
     MISSING_REQUIREMENTS = False
 except ImportError:
@@ -141,16 +143,23 @@ class JLinkBinaryRunner(ZephyrBinaryRunner):
         # to load the shared library distributed with the tools, which
         # provides an API call for getting the version.
         if not hasattr(self, '_jlink_version'):
+            # pylink >= 0.14.0 exposes JLink SDK DLL (libjlinkarm) in
+            # JLINK_SDK_STARTS_WITH, while previous versions use JLINK_SDK_NAME
+            if Version(pylink.__version__) >= Version('0.14.0'):
+                sdk = Library.JLINK_SDK_STARTS_WITH
+            else:
+                sdk = Library.JLINK_SDK_NAME
+
             plat = sys.platform
             if plat.startswith('win32'):
                 libname = Library.get_appropriate_windows_sdk_name() + '.dll'
             elif plat.startswith('linux'):
-                libname = Library.JLINK_SDK_NAME + '.so'
+                libname = sdk + '.so'
             elif plat.startswith('darwin'):
-                libname = Library.JLINK_SDK_NAME + '.dylib'
+                libname = sdk + '.dylib'
             else:
                 self.logger.warning(f'unknown platform {plat}; assuming UNIX')
-                libname = Library.JLINK_SDK_NAME + '.so'
+                libname = sdk + '.so'
 
             lib = Library(dllpath=os.fspath(Path(self.commander).parent /
                                             libname))

--- a/scripts/west_commands/runners/jlink.py
+++ b/scripts/west_commands/runners/jlink.py
@@ -7,7 +7,6 @@
 import argparse
 import logging
 import os
-from packaging.version import Version
 from pathlib import Path
 import shlex
 import subprocess
@@ -143,9 +142,9 @@ class JLinkBinaryRunner(ZephyrBinaryRunner):
         # to load the shared library distributed with the tools, which
         # provides an API call for getting the version.
         if not hasattr(self, '_jlink_version'):
-            # pylink >= 0.14.0 exposes JLink SDK DLL (libjlinkarm) in
-            # JLINK_SDK_STARTS_WITH, while previous versions use JLINK_SDK_NAME
-            if Version(pylink.__version__) >= Version('0.14.0'):
+            # pylink 0.14.0/0.14.1 exposes JLink SDK DLL (libjlinkarm) in
+            # JLINK_SDK_STARTS_WITH, while other versions use JLINK_SDK_NAME
+            if pylink.__version__ in ('0.14.0', '0.14.1'):
                 sdk = Library.JLINK_SDK_STARTS_WITH
             else:
                 sdk = Library.JLINK_SDK_NAME

